platform/darwin/src/MGLRasterSource.mm

```diff
@@ -37,76 +33,32 @@ - (instancetype)initWithIdentifier:(NSString *)identifier configurationURL:(NSUR
 }
 
 - (instancetype)initWithIdentifier:(NSString *)identifier configurationURL:(NSURL *)configurationURL tileSize:(CGFloat)tileSize {
-    if (self = [super initWithIdentifier:identifier configurationURL:configurationURL]) {
-        auto source = std::make_unique<mbgl::style::RasterSource>(identifier.UTF8String,
-                                                                  configurationURL.mgl_URLByStandardizingScheme.absoluteString.UTF8String,
-                                                                  uint16_t(round(tileSize)));
-        _pendingSource = std::move(source);
-        self.rawSource = _pendingSource.get();
-    }
-    return self;
+    auto source = std::make_unique<mbgl::style::RasterSource>(identifier.UTF8String,
+                                                              configurationURL.mgl_URLByStandardizingScheme.absoluteString.UTF8String,
+                                                              uint16_t(round(tileSize)));
+    return [super initWithPendingSource:std::move(source)];
```

Comment:
If you wanted to, you could say `return self = [super …]`, but the `if` is a convention that makes it easier to add and initialize ivars later on. The important part is assigning `self` and returning it; otherwise, ARC is free to assume that `self` is no longer in use and can release the object. (Along these lines, if you’re partway through the `if` block and discover that the object can’t be initialized for some reason, setting `self` to `nil` has the same effect of releasing the object.)